### ISSUE TITLE
Unable to resolve module missing-asset-registry-path

### ISSUE THREAD

**Issue by MajidAziz3** [facebook/react-native#38282]
### Description

error Unable to resolve module missing-asset-registry-path from path/src/assets/image.png: missing-asset-registry-path could not be found within the project or in these directories:
  node_modules
> 1 | �PNG
  2 | 
  3 | 
  4 | IHDRX^;I�sRGB���@IDATx�]  |E֯���$܇�C<�UBeUTċ@D/�k=�UDv�o��..
Error: Unable to resolve module missing-asset-registry-path from path/src/assets/image.png: missing-asset-registry-path could not be found within the project or in these directories:

### React Native Version

0.72.1

### Output of `npx react-native info`

System:
  OS: macOS 13.4
  CPU: (4) x64 Intel(R) Core(TM) i5-7360U CPU @ 2.30GHz
  Memory: 52.41 MB / 8.00 GB
  Shell:
    version: "5.9"
    path: /bin/zsh
Binaries:
  Node:
    version: 20.2.0
    path: /usr/local/bin/node
  Yarn:
    version: 1.22.19
    path: /usr/local/bin/yarn
  npm:
    version: 9.6.6
    path: /usr/local/bin/npm
  Watchman:
    version: 2023.05.22.00
    path: /usr/local/bin/watchman
Managers:
  CocoaPods:
    version: 1.12.1
    path: /Users/apple/.rubies/ruby-3.2.2/bin/pod
SDKs:
  iOS SDK:
    Platforms:
      - DriverKit 22.4
      - iOS 16.4
      - macOS 13.3
      - tvOS 16.4
      - watchOS 9.4
  Android SDK: Not Found
IDEs:
  Android Studio: 2022.2 AI-222.4459.24.2221.9971841
  Xcode:
    version: 14.3.1/14E300c
    path: /usr/bin/xcodebuild
Languages:
  Java:
    version: 17.0.7
    path: /usr/bin/javac
  Ruby:
    version: 3.2.2
    path: /Users/apple/.rubies/ruby-3.2.2/bin/ruby
npmPackages:
  "@react-native-community/cli":
    installed: 11.3.5
    wanted: ^11.3.5
  react:
    installed: 18.2.0
    wanted: 18.2.0
  react-native:
    installed: 0.72.1
    wanted: 0.72.1
  react-native-macos: Not Found
npmGlobalPackages:
  "*react-native*": Not Found
Android:
  hermesEnabled: true
  newArchEnabled: false
iOS:
  hermesEnabled: true
  newArchEnabled: false

### Steps to reproduce

Error coming while running Apk bundling commands npx react-native bundle --platform android --dev false --entry-file index.js --bundle-output android/app/src/main/assets/index.android.bundle --assets-dest android/app/src/main/res

### Snack, code example, screenshot, or link to a repository

command for bundling assets: npx react-native bundle --platform android --dev false --entry-file index.js --bundle-output android/app/src/main/assets/index.android.bundle --assets-dest android/app/src/main/res

**Comment by blakef** [comment]
Hi, we're going to need a bit more information to be helpful.  Here are some things you can do in the interim to unblock yourself:
- list your installed packages please, and confirm you're using the new `@react-native/metro-config` (might be worth reading @huntie's comment here: https://github.com/facebook/react-native/issues/36730#issuecomment-1491622539).
- take a look at the upgrade helper for the version you're moving to: https://react-native-community.github.io/upgrade-helper/?from=0.71.12&to=0.72.1

If you think this is a more general issue and not specific to your setup, can you provide a minimum reproducible example?

**Comment by MajidAziz3** [comment]
@blakef the project is too old , i am sure the issue is an metro.config , here is the config file , please have a look ,the issue is only with png images
 module.exports = (async () => {
   const {
     resolver: { sourceExts, assetExts },
   } = await getDefaultConfig();
 
   return {
     transformer: {
       babelTransformerPath: require.resolve('react-native-svg-transformer'),
       getTransformOptions: async () => ({
         transform: {
           experimentalImportSupport: false,
           inlineRequires: false,
         },
       }),
     },
     resolver: {
       assetExts: assetExts.filter(ext => ext !== 'svg'),
       sourceExts: [...sourceExts, 'svg'],
     },
   };
 })();
react-native Config File
module.exports = {
    project: {
      ios: {},
      android: {}, // grouped into "project"
    },
    assets: ["./src/assets/fonts/"], // stays the same
  };

**Comment by cortinico** [comment]
Potentially related to https://github.com/react-navigation/react-navigation/issues/11192

@MajidAziz3 without a repro we can't really help further.

**Comment by huntie** [comment]
Duplicate of https://github.com/facebook/metro/issues/1028.

**Comment by huntie** [comment]
Duplicate of https://github.com/facebook/metro/issues/1028.

**Comment by KushalDesai1** [comment]
I had to downgrade react-native version from 0.73 to 0.68.

And node version from v20 to v16.20.0.

**Comment by Leslie-Wong-H** [comment]
I am facing this with jpg, yet png is fine.
react-native 0.72.8

**Comment by Leslie-Wong-H** [comment]
> I am facing this with jpg, yet png is fine. react-native 0.72.8

This config works. 

```
/**
 * Metro configuration for React Native
 * https://github.com/facebook/react-native
 *
 * @format
 */

const { getDefaultConfig, mergeConfig } = require('@react-native/metro-config');

const defaultConfig = getDefaultConfig(__dirname);

const {
  resolver: { sourceExts, assetExts },
} = getDefaultConfig(__dirname);

const config = {
  transformer: {
    getTransformOptions: async () => ({
      transform: {
        experimentalImportSupport: false,
        inlineRequires: true,
      },
    }),
    babelTransformerPath: require.resolve('react-native-svg-transformer'),
  },
  resolver: {
    assetExts: assetExts.filter(ext => ext !== 'svg'),
    sourceExts: [...sourceExts, 'svg'],
  },
};

module.exports = mergeConfig(defaultConfig, config);

```

**Comment by leorodriguesdev** [comment]
@Leslie-Wong-H thank you, you saved me

**Comment by santiagoquintero95** [comment]
> > I am facing this with jpg, yet png is fine. react-native 0.72.8
> 
> This config works.
> 
> ```
> /**
>  * Metro configuration for React Native
>  * https://github.com/facebook/react-native
>  *
>  * @format
>  */
> 
> const { getDefaultConfig, mergeConfig } = require('@react-native/metro-config');
> 
> const defaultConfig = getDefaultConfig(__dirname);
> 
> const {
>   resolver: { sourceExts, assetExts },
> } = getDefaultConfig(__dirname);
> 
> const config = {
>   transformer: {
>     getTransformOptions: async () => ({
>       transform: {
>         experimentalImportSupport: false,
>         inlineRequires: true,
>       },
>     }),
>     babelTransformerPath: require.resolve('react-native-svg-transformer'),
>   },
>   resolver: {
>     assetExts: assetExts.filter(ext => ext !== 'svg'),
>     sourceExts: [...sourceExts, 'svg'],
>   },
> };
> 
> module.exports = mergeConfig(defaultConfig, config);
> ```

Omg finally it works. I was using an old config and I moved to react native 7.3.0  and this update is required or the archive will fail

**Comment by ShivBhanKushWaha** [comment]
error: Error: Unable to resolve module missing-asset-registry-path from D:\Hoverin Aerospace\HoverinApp\node_modules\react-native\Libraries\LogBox\UI\LogBoxImages\close.png: missing-asset-registry-path could not be found within the project or in these directories:
  node_modules\react-native\node_modules
  node_modules
> 1 | �PNG
  2 | →
  3 |
  4 | IHDR∟♦�♦↨▼�IDATxڝՅ
    at ModuleResolver.resolveDependency (D:\Hoverin Aerospace\HoverinApp\node_modules\metro\src\node-haste\DependencyGraph\ModuleResolution.js:153:15)
    at DependencyGraph.resolveDependency (D:\Hoverin Aerospace\HoverinApp\node_modules\metro\src\node-haste\DependencyGraph.js:271:43)
    at D:\Hoverin Aerospace\HoverinApp\node_modules\metro\src\lib\transformHelpers.js:176:21
    at resolveDependencies (D:\Hoverin Aerospace\HoverinApp\node_modules\metro\src\DeltaBundler\buildSubgraph.js:56:25)
    at visit (D:\Hoverin Aerospace\HoverinApp\node_modules\metro\src\DeltaBundler\buildSubgraph.js:107:30)
    at async Promise.all (index 12)
    at async visit (D:\Hoverin Aerospace\HoverinApp\node_modules\metro\src\DeltaBundler\buildSubgraph.js:116:5)
    at async Promise.all (index 5)
    at async visit (D:\Hoverin Aerospace\HoverinApp\node_modules\metro\src\DeltaBundler\buildSubgraph.js:116:5)
    at async Promise.all (index 15)
how to fix this issue

**Comment by alexdarh** [comment]
> error: Error: Unable to resolve module missing-asset-registry-path from D:\Hoverin Aerospace\HoverinApp\node_modules\react-native\Libraries\LogBox\UI\LogBoxImages\close.png: missing-asset-registry-path could not be found within the project or in these directories: node_modules\react-native\node_modules node_modules
> 
> > 1 | �PNG
> > 2 | →
> > 3 |
> > 4 | IHDR∟♦�♦↨▼�IDATxڝՅ
> > at ModuleResolver.resolveDependency (D:\Hoverin Aerospace\HoverinApp\node_modules\metro\src\node-haste\DependencyGraph\ModuleResolution.js:153:15)
> > at DependencyGraph.resolveDependency (D:\Hoverin Aerospace\HoverinApp\node_modules\metro\src\node-haste\DependencyGraph.js:271:43)
> > at D:\Hoverin Aerospace\HoverinApp\node_modules\metro\src\lib\transformHelpers.js:176:21
> > at resolveDependencies (D:\Hoverin Aerospace\HoverinApp\node_modules\metro\src\DeltaBundler\buildSubgraph.js:56:25)
> > at visit (D:\Hoverin Aerospace\HoverinApp\node_modules\metro\src\DeltaBundler\buildSubgraph.js:107:30)
> > at async Promise.all (index 12)
> > at async visit (D:\Hoverin Aerospace\HoverinApp\node_modules\metro\src\DeltaBundler\buildSubgraph.js:116:5)
> > at async Promise.all (index 5)
> > at async visit (D:\Hoverin Aerospace\HoverinApp\node_modules\metro\src\DeltaBundler\buildSubgraph.js:116:5)
> > at async Promise.all (index 15)
> > how to fix this issue
same error

**Comment by PavelVisme** [comment]
same error after upgrading to expo50 
Unable to resolve "missing-asset-registry-path" from "../../node_modules/.pnpm/react-native@0.73.4_@babel+core@7.23.3_@babel+preset-env@7.23.3_react@18.2.0/node_modules/react-native/Libraries/LogBox/UI/LogBoxImages/close.png"
This is example with almost empty index.js file

**Comment by DeabitTech** [comment]
but why there are a lot of issues about the upgrading of expo 50? @expo please fix it

**Comment by uoon-dev** [comment]
(Expo SDK 50) Every solution didn't work for me, but I solved missing-asset-registry-path issue by below custom metro config.

```javascript
// Learn more https://docs.expo.io/guides/customizing-metro
const { getDefaultConfig } = require('expo/metro-config');

/** @type {import('expo/metro-config').MetroConfig} */
const config = getDefaultConfig(__dirname);

config.transformer = {
  ...config.transformer,
  babelTransformerPath: require.resolve('react-native-svg-transformer'),
  assetPlugins: ['expo-asset/tools/hashAssetFiles'],
  getTransformOptions: async () => ({
    transform: {
      experimentalImportSupport: false,
      inlineRequires: true,
    },
  }),
};
//
config.resolver = {
  ...config.resolver,
  assetExts: config.resolver.assetExts.filter((ext) => ext !== 'svg'),
  sourceExts: [...config.resolver.sourceExts, 'svg', 'd.ts'],
};

module.exports = config;

```

**Comment by Tusharatha7** [comment]
with png. none of the above metroconfig is working for. what can i do?

Unable to resolve "./assets/logo.png" from "screens\Startscreen.js"

**Comment by ch3njust1n** [comment]
> (Expo SDK 50) Every solution didn't work for me, but I solved missing-asset-registry-path issue by below custom metro config.
> 
> ```js
> // Learn more https://docs.expo.io/guides/customizing-metro
> const { getDefaultConfig } = require('expo/metro-config');
> 
> /** @type {import('expo/metro-config').MetroConfig} */
> const config = getDefaultConfig(__dirname);
> 
> config.transformer = {
>   ...config.transformer,
>   babelTransformerPath: require.resolve('react-native-svg-transformer'),
>   assetPlugins: ['expo-asset/tools/hashAssetFiles'],
>   getTransformOptions: async () => ({
>     transform: {
>       experimentalImportSupport: false,
>       inlineRequires: true,
>     },
>   }),
> };
> //
> config.resolver = {
>   ...config.resolver,
>   assetExts: config.resolver.assetExts.filter((ext) => ext !== 'svg'),
>   sourceExts: [...config.resolver.sourceExts, 'svg', 'd.ts'],
> };
> 
> module.exports = config;
> ```

This worked for me! Thank you!

**Comment by KStockton** [comment]
> > (Expo SDK 50) Every solution didn't work for me, but I solved missing-asset-registry-path issue by below custom metro config.
> > ```js
> > // Learn more https://docs.expo.io/guides/customizing-metro
> > const { getDefaultConfig } = require('expo/metro-config');
> > 
> > /** @type {import('expo/metro-config').MetroConfig} */
> > const config = getDefaultConfig(__dirname);
> > 
> > config.transformer = {
> >   ...config.transformer,
> >   babelTransformerPath: require.resolve('react-native-svg-transformer'),
> >   assetPlugins: ['expo-asset/tools/hashAssetFiles'],
> >   getTransformOptions: async () => ({
> >     transform: {
> >       experimentalImportSupport: false,
> >       inlineRequires: true,
> >     },
> >   }),
> > };
> > //
> > config.resolver = {
> >   ...config.resolver,
> >   assetExts: config.resolver.assetExts.filter((ext) => ext !== 'svg'),
> >   sourceExts: [...config.resolver.sourceExts, 'svg', 'd.ts'],
> > };
> > 
> > module.exports = config;
> > ```
> 
> This worked for me! Thank you!

This worked for me as well.

**Comment by haidernayyar** [comment]
> > I am facing this with jpg, yet png is fine. react-native 0.72.8
> 
> This config works.
> 
> ```
> /**
>  * Metro configuration for React Native
>  * https://github.com/facebook/react-native
>  *
>  * @format
>  */
> 
> const { getDefaultConfig, mergeConfig } = require('@react-native/metro-config');
> 
> const defaultConfig = getDefaultConfig(__dirname);
> 
> const {
>   resolver: { sourceExts, assetExts },
> } = getDefaultConfig(__dirname);
> 
> const config = {
>   transformer: {
>     getTransformOptions: async () => ({
>       transform: {
>         experimentalImportSupport: false,
>         inlineRequires: true,
>       },
>     }),
>     babelTransformerPath: require.resolve('react-native-svg-transformer'),
>   },
>   resolver: {
>     assetExts: assetExts.filter(ext => ext !== 'svg'),
>     sourceExts: [...sourceExts, 'svg'],
>   },
> };
> 
> module.exports = mergeConfig(defaultConfig, config);
> ```

This worked for me and remember to re-start the metro

**Comment by quotentiroler** [comment]
I cannot get over this issue for png files after updating to expo sdk 51 :(

**Comment by Bunnavoit** [comment]
> (Expo SDK 50) Every solution didn't work for me, but I solved missing-asset-registry-path issue by below custom metro config.
> 
> ```js
> // Learn more https://docs.expo.io/guides/customizing-metro
> const { getDefaultConfig } = require('expo/metro-config');
> 
> /** @type {import('expo/metro-config').MetroConfig} */
> const config = getDefaultConfig(__dirname);
> 
> config.transformer = {
>   ...config.transformer,
>   babelTransformerPath: require.resolve('react-native-svg-transformer'),
>   assetPlugins: ['expo-asset/tools/hashAssetFiles'],
>   getTransformOptions: async () => ({
>     transform: {
>       experimentalImportSupport: false,
>       inlineRequires: true,
>     },
>   }),
> };
> //
> config.resolver = {
>   ...config.resolver,
>   assetExts: config.resolver.assetExts.filter((ext) => ext !== 'svg'),
>   sourceExts: [...config.resolver.sourceExts, 'svg', 'd.ts'],
> };
> 
> module.exports = config;
> ```

do you found the solution of this error ? i got error like this too and my metro.config is also like you
+ error 
Unable to resolve "missing-asset-registry-path"

**Comment by meshboy88** [comment]
> (Expo SDK 50) Every solution didn't work for me, but I solved missing-asset-registry-path issue by below custom metro config.
> 
> ```js
> // Learn more https://docs.expo.io/guides/customizing-metro
> const { getDefaultConfig } = require('expo/metro-config');
> 
> /** @type {import('expo/metro-config').MetroConfig} */
> const config = getDefaultConfig(__dirname);
> 
> config.transformer = {
>   ...config.transformer,
>   babelTransformerPath: require.resolve('react-native-svg-transformer'),
>   assetPlugins: ['expo-asset/tools/hashAssetFiles'],
>   getTransformOptions: async () => ({
>     transform: {
>       experimentalImportSupport: false,
>       inlineRequires: true,
>     },
>   }),
> };
> //
> config.resolver = {
>   ...config.resolver,
>   assetExts: config.resolver.assetExts.filter((ext) => ext !== 'svg'),
>   sourceExts: [...config.resolver.sourceExts, 'svg', 'd.ts'],
> };
> 
> module.exports = config;
> ```

Bro, you made my day. Thanks, it worked for the png issue!

**Comment by Lonka-Pardhu** [comment]
> with png. none of the above metroconfig is working for. what can i do?
> 
> Unable to resolve "./assets/logo.png" from "screens\Startscreen.js"

+1

**Comment by skirianov** [comment]
> (Expo SDK 50) Every solution didn't work for me, but I solved missing-asset-registry-path issue by below custom metro config.
> 
> ```js
> // Learn more https://docs.expo.io/guides/customizing-metro
> const { getDefaultConfig } = require('expo/metro-config');
> 
> /** @type {import('expo/metro-config').MetroConfig} */
> const config = getDefaultConfig(__dirname);
> 
> config.transformer = {
>   ...config.transformer,
>   babelTransformerPath: require.resolve('react-native-svg-transformer'),
>   assetPlugins: ['expo-asset/tools/hashAssetFiles'],
>   getTransformOptions: async () => ({
>     transform: {
>       experimentalImportSupport: false,
>       inlineRequires: true,
>     },
>   }),
> };
> //
> config.resolver = {
>   ...config.resolver,
>   assetExts: config.resolver.assetExts.filter((ext) => ext !== 'svg'),
>   sourceExts: [...config.resolver.sourceExts, 'svg', 'd.ts'],
> };
> 
> module.exports = config;
> ```

saved my day, thanks! 

If someone is running turborepo, this config worked for me 

```
const { getDefaultConfig } = require('expo/metro-config');
const path = require('path');

const projectRoot = __dirname;
const monorepoRoot = path.resolve(projectRoot, '../..');

const config = getDefaultConfig(projectRoot);

// Only list the packages within your monorepo that your app uses. No need to add anything else.
// If your monorepo tooling can give you the list of monorepo workspaces linked
// in your app workspace, you can automate this list instead of hardcoding them.
const monorepoPackages = {
  '@wptp/db': path.resolve(monorepoRoot, 'packages/db'),
};

// 1. Watch the local app directory, and only the shared packages (limiting the scope and speeding it up)
// Note how we change this from `monorepoRoot` to `projectRoot`. This is part of the optimization!
config.watchFolders = [projectRoot, ...Object.values(monorepoPackages)];

// Add the monorepo workspaces as `extraNodeModules` to Metro.
// If your monorepo tooling creates workspace symlinks in the `node_modules` directory,
// you can either add symlink support to Metro or set the `extraNodeModules` to avoid the symlinks.
// See: https://metrobundler.dev/docs/configuration/#extranodemodules
config.resolver.extraNodeModules = monorepoPackages;
config.transformer = {
  ...config.transformer,
  assetPlugins: ['expo-asset/tools/hashAssetFiles'],
  getTransformOptions: async () => ({
    transform: {
      experimentalImportSupport: false,
      inlineRequires: true,
    },
  }),
};

// 2. Let Metro know where to resolve packages and in what order
config.resolver.nodeModulesPaths = [
  path.resolve(projectRoot, 'node_modules'),
  path.resolve(monorepoRoot, 'node_modules'),
];

config.resolver = {
  ...config.resolver,
  assetExts: config.resolver.assetExts.filter((ext) => ext !== 'svg'),
  sourceExts: [...config.resolver.sourceExts, 'svg', 'd.ts'],
};

module.exports = config;
```

**Comment by SATYAJIT-techi** [comment]
> > I am facing this with jpg, yet png is fine. react-native 0.72.8
> 
> This config works.
> 
> ```
> /**
>  * Metro configuration for React Native
>  * https://github.com/facebook/react-native
>  *
>  * @format
>  */
> 
> const { getDefaultConfig, mergeConfig } = require('@react-native/metro-config');
> 
> const defaultConfig = getDefaultConfig(__dirname);
> 
> const {
>   resolver: { sourceExts, assetExts },
> } = getDefaultConfig(__dirname);
> 
> const config = {
>   transformer: {
>     getTransformOptions: async () => ({
>       transform: {
>         experimentalImportSupport: false,
>         inlineRequires: true,
>       },
>     }),
>     babelTransformerPath: require.resolve('react-native-svg-transformer'),
>   },
>   resolver: {
>     assetExts: assetExts.filter(ext => ext !== 'svg'),
>     sourceExts: [...sourceExts, 'svg'],
>   },
> };
> 
> module.exports = mergeConfig(defaultConfig, config);
> ```

make sure to start metro with --reset-cache

**Comment by hissamumair** [comment]
> > > I am facing this with jpg, yet png is fine. react-native 0.72.8
> > 
> > 
> > This config works.
> > ```
> > /**
> >  * Metro configuration for React Native
> >  * https://github.com/facebook/react-native
> >  *
> >  * @format
> >  */
> > 
> > const { getDefaultConfig, mergeConfig } = require('@react-native/metro-config');
> > 
> > const defaultConfig = getDefaultConfig(__dirname);
> > 
> > const {
> >   resolver: { sourceExts, assetExts },
> > } = getDefaultConfig(__dirname);
> > 
> > const config = {
> >   transformer: {
> >     getTransformOptions: async () => ({
> >       transform: {
> >         experimentalImportSupport: false,
> >         inlineRequires: true,
> >       },
> >     }),
> >     babelTransformerPath: require.resolve('react-native-svg-transformer'),
> >   },
> >   resolver: {
> >     assetExts: assetExts.filter(ext => ext !== 'svg'),
> >     sourceExts: [...sourceExts, 'svg'],
> >   },
> > };
> > 
> > module.exports = mergeConfig(defaultConfig, config);
> > ```
> 
> Omg finally it works. I was using an old config and I moved to react native 7.3.0 and this update is required or the archive will fail

thank you so much it work

**Comment by PriyanshuPandey-github** [comment]
I've similar issue I'm using react-native cli not expo. I was upgrading my app with the help of react-native-upgrade helper but whenever i switch the version from 0.73.11 to 0.74.0 the `npx react-native config` return `"project": {"ios": null}"` and it does not let me install pods saying undefined method for nil class in `use_native_modules!`

**Comment by carsonxw** [comment]
> (Expo SDK 50) Every solution didn't work for me, but I solved missing-asset-registry-path issue by below custom metro config.
> 
> ```js
> // Learn more https://docs.expo.io/guides/customizing-metro
> const { getDefaultConfig } = require('expo/metro-config');
> 
> /** @type {import('expo/metro-config').MetroConfig} */
> const config = getDefaultConfig(__dirname);
> 
> config.transformer = {
>   ...config.transformer,
>   babelTransformerPath: require.resolve('react-native-svg-transformer'),
>   assetPlugins: ['expo-asset/tools/hashAssetFiles'],
>   getTransformOptions: async () => ({
>     transform: {
>       experimentalImportSupport: false,
>       inlineRequires: true,
>     },
>   }),
> };
> //
> config.resolver = {
>   ...config.resolver,
>   assetExts: config.resolver.assetExts.filter((ext) => ext !== 'svg'),
>   sourceExts: [...config.resolver.sourceExts, 'svg', 'd.ts'],
> };
> 
> module.exports = config;
> ```

Cool, still works on 0.76.5. Thanks!

**Comment by detailed671** [comment]
> Hi, we're going to need a bit more information to be helpful. Here are some things you can do in the interim to unblock yourself:
> 
> * list your installed packages please, and confirm you're using the new `@react-native/metro-config` (might be worth reading [@huntie](https://github.com/huntie)'s comment here: [Cant bundle locally on port 8080 after init on 0.72.0-rc.0 and bug in the template #36730 (comment)](https://github.com/facebook/react-native/issues/36730#issuecomment-1491622539)).
> * take a look at the upgrade helper for the version you're moving to: https://react-native-community.github.io/upgrade-helper/?from=0.71.12&to=0.72.1
> 
> If you think this is a more general issue and not specific to your setup, can you provide a minimum reproducible example?





https://hypertexter.space/questions/only-one-admin-account-can-be-accessed-at-a-time-by-multiple-logins

**Comment by cvincentcoleman** [comment]
What ultimately worked for me was 

`npm i react-native/assets-registry`

And this metro config

 ```
const {getDefaultConfig} = require('expo/metro-config');
const {
  mergeConfig,
  getDefaultConfig: rnGetDefaultConfig,
} = require('@react-native/metro-config');

/**
 * Metro configuration
 * https://facebook.github.io/metro/docs/configuration
 *
 * @type {import('metro-config').MetroConfig}
 */
const config = {
  transformer: {
    ...rnGetDefaultConfig().transformer,
    assetPlugins: ['expo-asset/tools/hashAssetFiles'],
  },
};

module.exports = mergeConfig(getDefaultConfig(__dirname), config);
```

**Comment by georgebezerra** [comment]
Hi,

I've been trying to solve this problem for 3 days and I can't. I've tried several solutions I found on the internet, and I've installed react-native/assets-registry but to no avail.

And I can't downgrade react-native or install a newer version, I have to work with 0.72.*

If anyone has any tips, I'd appreciate it. 

I'm sharing my project configuration:

1. metro config:
```
import path from 'path';
import { getDefaultConfig, mergeConfig } from '@react-native/metro-config';
import { fileURLToPath } from 'url';

const __filename = fileURLToPath(import.meta.url); // get the resolved path to the file
const __dirname = path.dirname(__filename); // get the name of the directory
const defaultConfig = getDefaultConfig(__dirname);
const { resolver: { sourceExts, assetExts } } = defaultConfig;

const config = {
  transformer: {
    getTransformOptions: async () => ({
      transform: {
        experimentalImportSupport: false,
        inlineRequires: true,
      },
    }),
    babelTransformerPath: ("react-native-svg-transformer"),
  },
  resolver: {
    assetExts: [...assetExts, 'png', 'jpg', 'jpeg', 'bin'],
    sourceExts: [...sourceExts, 'jsx', 'js', 'ts', 'tsx', 'd.ts', 'svg'],
    resolverMainFields: ["sbmodern", "react-native", "browser", "main"],
    unstable_enablePackageExports: true,
  },
  watchFolders: [path.resolve(__dirname)],
};

export default mergeConfig(defaultConfig, config);
```

2. babel.config.js
```
export const presets = [
  "module:metro-react-native-babel-preset"
];
````

3. package.json
```
{
  "name": "MyApp",
  "private": true,
  "type": "module",
  "version": "0.0.1",
  "scripts": {
    "android": "react-native run-android",
    "ios": "react-native run-ios",
    "lint": "eslint .",
    "start": "react-native start --reset-cache --port=8081",
    "test": "jest",
    "clean:cache": "yarn cache clean && npm cache clean --force",
    "clean:node": "rm -rf node_modules yarn.lock package-lock.json"
  },
  "dependencies": {
    "@svgr/core": "8.1.0",
    "path-dirname": "1.0.2",
    "react": "18.2.0",
    "react-native": "0.72.17",
    "react-native-svg": "15.0.0",
    "ts-node": "10.9.2"
  },
  "devDependencies": {
    "@babel/core": "7.26.10",
    "@babel/preset-env": "7.26.9",
    "@babel/runtime": "7.27.0",
    "@react-native/assets-registry": "0.79.1",
    "@react-native/eslint-config": "0.72.2",
    "@react-native/js-polyfills": "0.79.1",
    "@react-native/metro-babel-transformer": "0.79.1",
    "@react-native/metro-config": "0.72.12",
    "@svgr/plugin-jsx": "8.1.0",
    "@svgr/plugin-svgo": "8.1.0",
    "@tsconfig/react-native": "3.0.0",
    "@types/react": "18.0.24",
    "@types/react-test-renderer": "18.0.0",
    "babel-jest": "29.2.1",
    "eslint": "8.19.0",
    "jest": "29.2.1",
    "metro-config": "0.82.1",
    "metro-react-native-babel-preset": "0.77.0",
    "metro-runtime": "0.82.1",
    "prettier": "2.4.1",
    "react-native-svg-transformer": "1.5.0",
    "react-test-renderer": "18.2.0",
    "typescript": "4.8.4"
  },
  "engines": {
    "node": ">=20"
  }
}
```

I appreciate any help in advance!

**Comment by huntie** [comment]
@georgebezerra 

> And I can't downgrade react-native or install a newer version, I have to work with 0.72.*

Please downgrade all of your `@react-native/*` and `metro-react-native-babel-preset` dependencies to `0.72.*` to match.

**Comment by vaishnav-noovosoft** [comment]
1. Metro doesn't support async config exports - When you export (async () => {...})(), Metro receives a Promise instead of a configuration object
2. Asset registry couldn't initialize - Because Metro couldn't properly read the config, the asset registry path wasn't set up correctly
3. Result: "missing-asset-registry-path" error - When Metro tried to bundle your fonts (.ttf files), it couldn't find the asset registry to process them
